recipes/source.rb

```diff
@@ -1,47 +1,45 @@
 include_recipe 'build-essential'
 
-Array(node[:ruby_installer][:source_package_dependencies]).each do |pkg|
+Array(node['ruby_installer']['source_package_dependencies']).each do |pkg|
   package pkg
 end
 
-bash "install_ruby" do
-  cwd "/usr/src"
+bash 'install_ruby' do
+  cwd '/usr/src'
   code <<-EOH
-    tar -zxf ruby-#{node[:ruby_installer][:source_version]}.tar.gz
-    cd ruby-#{node[:ruby_installer][:source_version]}/
+    tar -zxf ruby-#{node['ruby_installer']['source_version']}.tar.gz
+    cd ruby-#{node['ruby_installer']['source_version']}/
     autoconf
-    #{
-      if(node[:ruby_installer][:source_falcon_patch]) 
-        "curl https://raw.github.com/gist/4136373/falcon-gc.diff | patch -p1"
-      end
-    }
-    ./configure --prefix=#{node[:ruby_installer][:source_install_dir]} --disable-install-doc --enable-shared
-    make -j#{node[:cpu][:total]+1}
+    #{if node['ruby_installer']['source_falcon_patch']
+        'curl https://raw.github.com/gist/4136373/falcon-gc.diff | patch -p1'
+      end}
+    ./configure --prefix=#{node['ruby_installer']['source_install_dir']} --disable-install-doc --enable-shared
+    make -j#{node['cpu']['total'] + 1}
     make install
   EOH
   action :nothing
-  notifies :reload, resources(:ohai => 'ruby'), :immediately
-  if(node[:ruby_installer][:source_optimize])
-    environment :cflags => "-march=native -pipe -fomit-frame-pointer -O#{node[:ruby_installer][:source_optimization_level]}"
+  notifies :reload, 'ohai[ruby]', :immediately
+  if node['ruby_installer']['source_optimize']
+    environment cflags: "-march=native -pipe -fomit-frame-pointer -O#{node['ruby_installer']['source_optimization_level']}"
   end
 end
 
-remote_file "/usr/src/ruby-#{node[:ruby_installer][:source_version]}.tar.gz" do
+remote_file "/usr/src/ruby-#{node['ruby_installer']['source_version']}.tar.gz" do
   not_if do
     begin
-      v = %{#{File.join(node[:ruby_installer][:source_install_dir], 'bin', 'ruby')} --version}.strip
-      v.split(' ')[1] == node[:ruby_installer][:source_version].sub('-', '')
+      v = %(#{File.join(node['ruby_installer']['source_install_dir'], 'bin', 'ruby')} --version).strip
+      v.split(' ')[1] == node['ruby_installer']['source_version'].sub('-', '')
     rescue Errno::ENOENT
       false
     end
   end
-  source "http://ftp.ruby-lang.org/pub/ruby/#{node[:ruby_installer][:source_version][0..2]}/ruby-#{node[:ruby_installer][:source_version]}.tar.gz"
+  source "http://ftp.ruby-lang.org/pub/ruby/#{node['ruby_installer']['source_version'][0..2]}/ruby-#{node['ruby_installer']['source_version']}.tar.gz"
   action :create_if_missing
-  notifies :run, "bash[install_ruby]", :immediately
+  notifies :run, 'bash[install_ruby]', :immediately
 end
 
-rin_ver = Gem::Version.new(node[:ruby_installer][:source_version].match(/(\d+\.?){2,3}/).to_s)
+rin_ver = Gem::Version.new(node['ruby_installer']['source_version'].match(/(\d+\.?){2,3}/).to_s)
```

Comment:
Mostly just thinking aloud here, but I wonder if there's a cleaner way to do this, possibly using the ruby Ohai data?